### PR TITLE
[candi] change helm tolerations template functions

### PR DESCRIPTION
**Pull request by RomanenkoDenys**
## Description

Simplify work with tolerations in deckhouse modules.
[Doc](https://github.com/deckhouse/deckhouse/wiki/Checklist-for-a-new-module#tolerations).
Related [PR](https://github.com/deckhouse/lib-helm/pull/5) in lib-helm.
## Why do we need it, and what problem does it solve?

We have problem to add tolerations in corner cases, so this PR simplify this process.
## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi
type: chore
summary: Changes helm tolerations template functions.
impact: All Pods will restart due to toleration changes.
impact_level: default
```

### CODE REVIEW

**Review by zuzzas — APPROVED** [review]
A thing of beauty.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by RomanenkoDenys** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.23 4304132440 4304132440-1-con-1-23 /sys/deckhouse-oss/install:pr3959 centos@3.125.207.212

**Comment by RomanenkoDenys** [comment]
/e2e/abort gcp;WithoutNAT;containerd;1.23 4304132295 4304132295-1-con-1-23 /sys/deckhouse-oss/install:pr3959 user@34.159.53.183

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/gcp'...

**Comment by z9r5** [comment]
/e2e/abort gcp;WithoutNAT;containerd;1.23 4306589596 4306589596-1-con-1-23 /sys/deckhouse-oss/install:pr3959 user@34.159.88.132

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by RomanenkoDenys** [comment]
/e2e/abort gcp;WithoutNAT;containerd;1.23 4306589596 4306589596-1-con-1-23 /sys/deckhouse-oss/install:pr3959 user@34.159.88.132

**Comment by RomanenkoDenys** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.23 4306589386 4306589386-1-con-1-23 /sys/deckhouse-oss/install:pr3959 centos@3.123.255.204

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've recognized your '/e2e/abort' command and started the workflow...

File: modules/500-upmeter/templates/smoke-mini/statefulsets.yaml [hunk 1]
```diff
@@ -73,7 +73,7 @@ spec:
                 operator: In
                 values:
                 - {{ $sts.node | quote }}
-      {{- include "helm_lib_tolerations" (tuple $context "any-node") | nindent 6 }}
+      {{- include "helm_lib_tolerations" (tuple $context "any-node" "without-storage-problems") | nindent 6 }}
```
Comment (shvgn) [inline modules/500-upmeter/templates/smoke-mini/statefulsets.yaml:76]:
```suggestion
{{- if ne $storageClass "false" }}
      {{- include "helm_lib_tolerations" (tuple $context "without-storage-problems" "without-storage-problems") | nindent 6 }}
{{- else}}
      {{- include "helm_lib_tolerations" (tuple $context "any-node" "without-storage-problems") | nindent 6 }}
{{- end }}
```

Comment (shvgn) [reply modules/500-upmeter/templates/smoke-mini/statefulsets.yaml:76]: In general, smoke-mini should always schedule on any node regardless of storage problems. synthetic/disk probe is disabled by default, so it's up to user to use the probe along with storage problems.

I suggest keeping "any-node".

File: ee/modules/600-flant-integration/templates/pricing/daemonset.yaml [hunk 2]
```diff
@@ -62,7 +62,7 @@ spec:
       - name: deckhouse-registry
       serviceAccountName: pricing
       {{- include "helm_lib_node_selector" (tuple . "master") | nindent 6 }}
-      {{- include "helm_lib_tolerations" (tuple . "master") | nindent 6 }}
+      {{- include "helm_lib_tolerations" (tuple . "any-node") | nindent 6 }}
```
Comment (shvgn) [inline ee/modules/600-flant-integration/templates/pricing/daemonset.yaml:65]: Why?

Comment (RomanenkoDenys) [reply ee/modules/600-flant-integration/templates/pricing/daemonset.yaml:65]: Because we do not have toleration master now. It was similar to any node.

**Review by shvgn — DISMISSED** [review]

**Review by yalosev — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/gcp'...

**Review by konstantin-axenov — APPROVED** [review]
